cmake/SociConfig.cmake

```diff
@@ -69,7 +69,7 @@ else()
             set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
         elseif (CMAKE_SYSTEM_NAME MATCHES "Windows")
             # We need to disable the warnings for using %I64d with MinGW.
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format")
+            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-pedantic-ms-format -Wno-format")
```

Comment:
@snikulov Obviously, we are not going to switch to C++11 soon, but I don't mind if we have private wrappers based on C++11 features for those troublemakers and we `#ifdef` activate them for MinGW exclusively.
On any other C++11 compilers, like VC++ and GCC, those wrappers can be activated with our CMake option `SOCI_CXX_C11`. By default, builds using VC++ and GCC should not require C++11.